src/main/java/gyro/aws/ecs/EcsServiceResource.java

```diff
@@ -0,0 +1,693 @@
+package gyro.aws.ecs;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.aws.iam.RoleResource;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.CollectionMax;
+import gyro.core.validation.ConflictsWith;
+import gyro.core.validation.DependsOn;
+import gyro.core.validation.Range;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import software.amazon.awssdk.services.ecs.EcsClient;
+import software.amazon.awssdk.services.ecs.model.CreateServiceRequest;
+import software.amazon.awssdk.services.ecs.model.CreateServiceResponse;
+import software.amazon.awssdk.services.ecs.model.DeploymentControllerType;
+import software.amazon.awssdk.services.ecs.model.LaunchType;
+import software.amazon.awssdk.services.ecs.model.PropagateTags;
+import software.amazon.awssdk.services.ecs.model.SchedulingStrategy;
+import software.amazon.awssdk.services.ecs.model.Tag;
+import software.amazon.awssdk.services.ecs.model.UpdateServiceRequest;
+
+/**
+ * Create an ECS service.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     aws::service example-service
+ *         name: "example-service"
+ *         task-definition: $(external-query aws::ecs-task-definition { family: 'test-fargate-task-definition', revision: 1 })
+ *         cluster: $(aws::ecs-cluster ecs-cluster-example)
+ *         desired-count: 1
+ *         enable-ecs-managed-tags: false
+ *         launch-type: EC2
+ *         scheduling-strategy: REPLICA
+ *
+ *         deployment-configuration
+ *             maximum-percent: 200
+ *             minimum-healthy-percent: 100
+ *         end
+ *
+ *         deployment-controller
+ *             type: ECS
+ *         end
+ *
+ *         network-configuration
+ *             aws-vpc-configuration
+ *                 assign-public-ip: DISABLED
+ *                 security-groups: [
+ *                     $(aws::security-group security-group)
+ *                 ]
+ *                 subnets: [
+ *                     $(aws::subnet "subnet-us-east-1a"),
+ *                     $(aws::subnet "subnet-us-east-1b")
+ *                 ]
+ *             end
+ *         end
+ *
+ *         placement-constraint
+ *             expression: "agentVersion > 0"
+ *             type: "memberOf"
+ *         end
+ *
+ *         placement-strategy
+ *             type: binpack
+ *             field: "MEMORY"
+ *         end
+ *     end
+ */
+@Type("service")
```

Comment:
Resource type name should be `ecs-service` as `service` is too generic, and we followed the pattern for the other resources as well. Update the the finder and all the examples as well.
```suggestion
@Type("ecs-service")
```